### PR TITLE
Switch panel

### PR DESCRIPTION
**Pull request by zuoyuanh**
Adding a description...this summer we have been working on a number of extensions that add new L sidebar tabs. The left sidebar it getting a bit crowded, so we wanted an easy way for users to move tabs back and forth between the R/L. Right now this PR adds:

Two new commands that will move the active widget from the L->R and R->L

In the view menu, this PR added two new items: 'Move Active Widget to Right Sidebar' (which moves the active widget in the left sidebar to the right side, if any) and 'Move Active Widget to Left Sidebar' (which moves the active widget in the right sidebar to the left side, if any)

The way this feature is implemented is when user click on the switch button, it will remove the designated widget from the source panel and add it to the destination panel. For each panel, there is an array that stores the id of the widgets moved by users. This array is stored by the LayoutRestorer and will be restored if user refresh the page. When the layout is being restored, it will restore the panel layout based on the information stored in the array.

### CODE REVIEW

**Comment by ellisonbg** [comment]
I can review this in the am, but would be good to get someones take who is familiar with the layout restorer APIs.

File: dev_mode/package.json [hunk 1]
```diff
@@ -51,6 +51,7 @@
     "@jupyterlab/mainmenu-extension": "^0.6.2",
     "@jupyterlab/markdownviewer-extension": "^0.17.2",
     "@jupyterlab/mathjax2-extension": "^0.5.2",
+    "@jupyterlab/metapackage": "^0.17.2",
```
Comment (blink1073) [inline dev_mode/package.json]: `metapackage` shouldn't be in this file

File: examples/notebook/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import * as React from 'react';
```
Comment (blink1073) [inline examples/notebook/src/index.tsx]: Is this meant to stay or was it for debugging?

**Comment by ellisonbg** [comment]
OK I had a look at this:

* I think that rather than track the left-moved and right-moved widgets, the settings registry should contain lists of the widgets that are currently in the left/right panel. The `addToLeftArea` and `addToRightArea` methods should add the widget and populate those lists. Moving a widget should simply change which list it is in. That way a user can both see where the widget was placed by the extension itself, and move it. The key is to track the current side of a widget, not the changes.
* The context menus, menus and commands all need more work from a UX perspective, we will have to talk about this.
* It does work, but I am worried that this to too messy and still in need of a bunch of work before it is ready to go.

One option would be to remove all the context menu stuff and expose it only as commands for 0.34. But we would need to have a really solid way of modeling the state in the settings system first though as that is a public API in the settings that will be more painful to change.

**Comment by blink1073** [comment]
My vote is to defer this PR until 0.35.

**Comment by ellisonbg** [comment]
@zuoyuanh today in the meeting we discussed this:

* The feel was that we want to move back to the layout restorer
* Going to be too much to get it into 0.34

Let's touch base later today about it.

**Comment by blink1073** [comment]
Closing in favor of #5347, thanks!

**Pull request closed without merging**